### ISSUE TITLE
Setup with Vue Native Cli does not work

### ISSUE THREAD

**Issue by ommehaz** [GeekyAnts/vue-native-core#87]
I am following the steps provided here -> https://vue-native.io/docs/installation.html
The commands I am running:
1. npm install -g vue-native-cli
2. vue-native init myproject // Initializes crna project
3. cd myproject
4. npm install
5. npm start:
5a. starts metro bundler with text:

INFO
23:49
Starting Metro Bundler on port 19001.
INFO
23:49
Metro Bundler ready.

6. I scan the Qr code with my iphone camera.
6.a expo app opens and the red screen with error on is there and my metro bundler writes:

ERROR
23:52
Unable to resolve "../../App" from "node_modules/expo/AppEntry.js"
ERROR
23:52
Building JavaScript bundle: error

Content of my rn-cli.config.js (autogenerated):
________________________________________________________
```
module.exports = {
  getTransformModulePath() {
    return require.resolve("./vueTransformerPlugin.js");
  },
  getSourceExts() {
    return ["vue"];
  }
};
```
________________________________________________________

Content of my vueTransformerPlugin.js (autogenerated):
________________________________________________________
```
var semver = require('semver');
var vueNaiveScripts = require("vue-native-scripts");
var reactNativeVersionString = require("react-native/package.json").version;
var reactNativeMinorVersion = semver(reactNativeVersionString).minor;

if (reactNativeMinorVersion >= 56) {
  upstreamTransformer = require("metro/src/reactNativeTransformer");
} else if (reactNativeMinorVersion >= 52) {
  upstreamTransformer = require("metro/src/transformer");
} else if (reactNativeMinorVersion >= 47) {
  upstreamTransformer = require("metro-bundler/src/transformer");
} else if (reactNativeMinorVersion === 46) {
  upstreamTransformer = require("metro-bundler/build/transformer");
} else {
  // handle RN <= 0.45
  var oldUpstreamTransformer = require("react-native/packager/transformer");
  upstreamTransformer = {
    transform({ src, filename, options }) {
      return oldUpstreamTransformer.transform(src, filename, options);
    }
  };
}
var vueExtensions = ["vue"]; // <-- Add other extensions if needed.

module.exports.transform = function({ src, filename, options }) {
  if (vueExtensions.some(ext => filename.endsWith("." + ext))) {
    return vueNaiveScripts.transform({ src, filename, options });
  }
  return upstreamTransformer.transform({ src, filename, options });
};

```
______________________________________________________________

Content of node_modules/expo/AppEntry.js:
______________________________________________________________

```
import { KeepAwake, registerRootComponent } from 'expo';
import App from '../../App';

if (__DEV__) {
  KeepAwake.activate();
}

registerRootComponent(App);
```
______________________________________________________________

My package.json file:
______________________________________________________________
```
{
  "name": "empty-project-template",
  "main": "node_modules/expo/AppEntry.js",
  "private": true,
  "scripts": {
    "start": "expo start",
    "android": "expo start --android",
    "ios": "expo start --ios",
    "eject": "expo eject"
  },
  "dependencies": {
    "expo": "^30.0.1",
    "react": "16.3.1",
    "react-native": "https://github.com/expo/react-native/archive/sdk-30.0.0.tar.gz",
    "vue-native-core": "0.0.8",
    "vue-native-helper": "0.0.9"
  },
  "devDependencies": {
    "vue-native-scripts": "0.0.14"
  }
}
```
______________________________________________________

My computer is a macbook 15 from 2018 and my OS is macOS High Sierra version 10.13.6
What am I doing wrong?

**Comment by anekix** [comment]
experiencing the same issue. why is this happening?

**Comment by neeraj-singh47** [comment]
React-native-scripts has been deprecated. You'll have to add this line ```"packagerOpts": {
      "config": "rn-cli.config.js"
    }``` in app.json within expo object.

**Comment by neeraj-singh47** [comment]
We'll release a new vue-native-cli version containing the above fix

**Comment by zek** [comment]
> React-native-scripts has been deprecated. You'll have to add this line `"packagerOpts": { "config": "rn-cli.config.js" }` in app.json within expo object.

Actually I followed everything but still I get same error.

**Comment by neeraj-singh47** [comment]
@zek Did you restart the app again?

**Comment by zek** [comment]
You mean in the simulator? Yes I did. Also I restarted expo.

**Comment by neeraj-singh47** [comment]
@zek Can you verify that you have a rn-cli.config.js file in root directory?

**Comment by zek** [comment]
Yes I have. I tried to switch vue-native from my existing project. 
~~~
module.exports = {
  getTransformModulePath() {
    return require.resolve("./vueTransformerPlugin.js");
  },
  getSourceExts() {
    return ["vue"];
  }
};
~~~


~~~
{
  "name": "empty-project-template",
  "main": "node_modules/expo/AppEntry.js",
  "private": true,
  "scripts": {
    "start": "expo start",
    "android": "expo start --android",
    "ios": "expo start --ios",
    "eject": "expo eject"
  },
  "dependencies": {
    "connectycube-reactnative": "^1.3.3",
    "expo": "^30.0.1",
    "react": "16.3.1",
    "react-native": "https://github.com/expo/react-native/archive/sdk-30.0.0.tar.gz",
    "react-redux": "^5.0.7",
    "redux": "^4.0.0",
    "vue-native-core": "0.0.8",
    "vue-native-helper": "0.0.9"
  },
  "devDependencies": {
    "vue-native-scripts": "0.0.14"
  }
}
~~~

~~~
var semver = require('semver');
var vueNaiveScripts = require("vue-native-scripts");
var reactNativeVersionString = require("react-native/package.json").version;
var reactNativeMinorVersion = semver(reactNativeVersionString).minor;

if (reactNativeMinorVersion >= 56) {
	upstreamTransformer = require("metro/src/reactNativeTransformer");
} else if (reactNativeMinorVersion >= 52) {
	upstreamTransformer = require("metro/src/transformer");
} else if (reactNativeMinorVersion >= 47) {
	upstreamTransformer = require("metro-bundler/src/transformer");
} else if (reactNativeMinorVersion === 46) {
	upstreamTransformer = require("metro-bundler/build/transformer");
} else {
	// handle RN <= 0.45
	var oldUpstreamTransformer = require("react-native/packager/transformer");
	upstreamTransformer = {
		transform({ src, filename, options }) {
			return oldUpstreamTransformer.transform(src, filename, options);
		}
	};
}
var vueExtensions = ["vue"]; // <-- Add other extensions if needed.

module.exports.transform = function({ src, filename, options }) {
	if (vueExtensions.some(ext => filename.endsWith("." + ext))) {
		return vueNaiveScripts.transform({ src, filename, options });
	}
	return upstreamTransformer.transform({ src, filename, options });
};
~~~

**Comment by neeraj-singh47** [comment]
@zek Do you already have an existing project? If yes then what is it based on?

**Comment by zek** [comment]
Just recently created react native project with connectycube.

**Comment by neeraj-singh47** [comment]
@zek What exactly is the error? Can you post the screenshot?

**Comment by zek** [comment]
Same error what @ommehaz gets.

Unable to resolve "../../App" from "node_modules/expo/AppEntry.js"

I just deleted node_modules directory then run `npm install` 

now it gives the error below 
<img width="466" alt="screen shot 2018-10-08 at 22 46 48" src="https://user-images.githubusercontent.com/3463291/46630178-19774500-cb4c-11e8-8f50-333c3953934e.png">

**Comment by neeraj-singh47** [comment]
@zek I assume you did the below steps again 
`cd node_modules/connectycube-reactnative/xmpp.js
make
cd ../../../
cp -r node_modules/connectycube-reactnative/xmpp.js/node_modules/* node_modules/` as specified in connectycube

**Comment by zek** [comment]
yes without vue-native it works. But I wanna use vue-native and connectycube together.

**Comment by zek** [comment]
<img width="433" alt="screen shot 2018-10-08 at 22 54 47" src="https://user-images.githubusercontent.com/3463291/46630545-2e080d00-cb4d-11e8-9874-14493183b862.png">

**Comment by zek** [comment]
@neeraj-singh47 I just renamed App.vue to App.js and tried, it failed then changed it to App.vue again now it works. Maybe it was a caching problem but its fine. Thank you 👍

**Comment by PsychoXIVI** [comment]
I have same issue. :C

**Comment by guillaumeduhan** [comment]
Hello,

Same issue here. After following solutions above, couldn’t resolve this error : « Unable to resolve module App... ».

I removed node modules, uninstall and install again crna, Vue-native and even watchman from Facebook.

Error remain the same.

**Comment by gabfr** [comment]
@guillaumeduhan @PsychoXIVI  Someone managed to get around this issue?
I already tried the steps above and nothing worked

**Comment by minsulander** [comment]
+1 same issue here

**Comment by gabfr** [comment]
> @guillaumeduhan @PsychoXIVI Someone managed to get around this issue?
> I already tried the steps above and nothing worked

Both methods: creating the project with the vue native cli, and also creating with the react native cli and then installing the vue-native packages result essentially in the same error

**Comment by neeraj-singh47** [comment]
Have you guys tried out https://github.com/GeekyAnts/vue-native-core/issues/101#issuecomment-437606353?

**Comment by jimwmg** [comment]
> Latest version of Vue Native Cli with the fix has been released. https://github.com/GeekyAnts/vue-native-cli/releases/tag/0.0.3

still not work  ,same error Unable to resolve "../../App" from "node_modules/expo/AppEntry.js"
E

**Comment by jimwmg** [comment]
you can try this to solve the problem;
[try this ](https://github.com/GeekyAnts/vue-native-core/issues/91#issuecomment-431216570)

**Comment by guillaumeduhan** [comment]
hello @gabfr no solution founded and tried everything before

**Comment by guillaumeduhan** [comment]
Hello @zek you succeed by changing app.vue in app.js and again in app.vue by refreshing? Or restart server?

**Comment by manusiakemos** [comment]
still waiting

**Comment by neeraj-singh47** [comment]
There's a problem with latest version of expo. We'll release a fix soon.

**Comment by manusiakemos** [comment]
yo what's up

**Comment by lopans** [comment]
> There's a problem with latest version of expo. We'll release a fix soon.

that's awesome, but how long? And is there any hot local fix?

**Comment by croossin** [comment]
@neeraj-singh47 Any updates on the fix?

**Comment by croossin** [comment]
I was able to easily fix the issue locally with [this response](https://github.com/GeekyAnts/vue-native-core/issues/101#issuecomment-437606353).

**Comment by guillaumeduhan** [comment]
@croossin Hello friend, I am going to try it soon. Tell you if it's working for me

**Comment by guillaumeduhan** [comment]
Okay everybody, @croossin had [the solution before here](https://github.com/GeekyAnts/vue-native-core/issues/101#issuecomment-437606353).

**Comment by ProjectInfinity** [comment]
This is still a problem.

**Comment by hkvega01** [comment]
I have same problem but i can't resolve that

**Comment by milgron** [comment]
Still the same problem here.

**Comment by jjmahe** [comment]
yes, I confirm, still the same issue, with a fresh expo and vue-native install. neither blank or tabs default apps work. did the rm things, no joy. I don't want to touch config files as I don't want to do what I don't understand. should work as per instructions from the install scripts themselves.

**Comment by jjmahe** [comment]
in fact when trying the expo build:os it says that it is not looking for App.vue ! that is a pity when trying a vue-native build!

**Comment by moneil868** [comment]
Has there been any progress on a fix for this?

**Comment by Cipher7** [comment]
My project install and ran correctly after I installed yarn, I hope this helps someone.
